mitodl/open-discussions
# Save embedly thumbnail URL's

#### What are the relevant tickets?
Partially addresses #926 

#### What's this PR do?
- Adds a `LinkMeta` model with attributes `url` and `thumbnail`
- Adds a nullable ForeignKey from `Post` to `LinkMeta` (but this isn't really used for anything yet).
- Sets the `Post.link_meta` foreign key value for link posts in `channels.api.sync_post_model`
- Calls `LinkMeta.objects.get_or_create` every time the `embedly.api.views.embedly_view` is invoked (ie when creating or viewing a link post)
- Calls `LinkMeta.objects.get_or_create` on `PostSerializer.create` and adds `thumbnail` to the `channels.serializers.BasePostSerializer` class.
- Adds the thumbnail (`post_link_thumbnail`) to the `search.serializers.ESPostSerializer` class so that it is included in the index.

#### How should this be manually tested?
- Create some link posts and text posts.  There should not be any obvious changes on the front end.  
- Check the REST API response for link posts and text posts (`/api/v0/posts/<id>).  There should be a `thumbnail` attribute for both, that should always be null for text posts, but equal to whatever thumbnail was returned by embedly for link posts.  (They will also be null for old link posts created before this PR, until the post detail page is viewed or a new link post is made with the same URL).
- Do an elasticsearch query on your new link posts and make sure that `post_link_thumbnail` is populated with the correct values.
